fusionbackend/src/main/java/io/fusion/fusionbackend/service/FactorySiteService.java

```diff
@@ -34,18 +37,39 @@
     private final CompanyService companyService;
     private final RoomService roomService;
     private final CountryService countryService;
+    private final ShiftSettingsService shiftSettingsService;
+    private final ShiftSettingsRepository shiftSettingsRepository;
 
     @Autowired
-    public FactorySiteService(FactorySiteRepository factorySiteRepository, CompanyService companyService,
-                              @Lazy RoomService roomService, @Lazy CountryService countryService) {
+    public FactorySiteService(FactorySiteRepository factorySiteRepository,
+                              CompanyService companyService,
+                              @Lazy RoomService roomService,
+                              @Lazy CountryService countryService,
+                              ShiftSettingsService shiftSettingsService,
+                              ShiftSettingsRepository shiftSettingsRepository) {
         this.factorySiteRepository = factorySiteRepository;
         this.companyService = companyService;
         this.roomService = roomService;
         this.countryService = countryService;
+        this.shiftSettingsService = shiftSettingsService;
+        this.shiftSettingsRepository = shiftSettingsRepository;
     }
 
     public Set<FactorySite> getFactorySitesByCompany(final Long companyId) {
-        return factorySiteRepository.findAllByCompanyId(FactorySiteRepository.DEFAULT_SORT, companyId);
+        Set<FactorySite> factorySites = factorySiteRepository
+                .findAllByCompanyId(FactorySiteRepository.DEFAULT_SORT, companyId);
+
+        for (FactorySite factorySite : factorySites) {
+            if (factorySite.hasShiftSettings()) {
+                ShiftSettings shiftSettings = shiftSettingsRepository
+                        .findDeepById(factorySite.getShiftSettings().getId())
+                        .orElseThrow();
+                factorySite.setShiftSettings(shiftSettings);
+            }
+
+        }
+
+        return factorySites;
```

Comment:
Maybe in a separate method or are shiftSettings always of interest?

Reply:
Would also recommend using an @Query with fetch joins on a specialized method in factorySiteRepository